### PR TITLE
[ENH] Add mrds scripts part 1

### PR DESCRIPTION
**Pull request by arnaudbore**
# Quick description

First MRDS script - use MRDS output and NUFO like image and select the expected solutions.

...
## Type of change

Check the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Provide data, screenshots, command line to test (if relevant)

[data](https://box.criugm.qc.ca/f/7869321dc89541b1bf3e/?dl=1)

Command line:

```
scil_mrds_select_number_of_tensors.py sub-01 nufo.nii.gz 
```
# Checklist

- [ ] My code follows the style guidelines of this project (run [autopep8](https://pypi.org/project/autopep8/))
- [ ] I added relevant citations to scripts, modules and functions docstrings and descriptions
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I moved all functions from the script file (except the argparser and main) to scilpy modules
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: scripts/scil_mrds_select_number_of_tensors.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Fit the info from a NuFO map to the plausible MRDS solutions.
```
Comment (mdesco) [inline scripts/scil_mrds_select_number_of_tensors.py:5]: Use the NUFO map information to select the plausible number of tensors in the MRDS solutions.

File: scripts/scil_mrds_select_number_of_tensors.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+Each MRDS input is a list of 5 files:
+    - Component size
+    - Eigenvalues
+    - Isotropic
+    - Number of components
+    - PDDs
```
Comment (mdesco) [inline scripts/scil_mrds_select_number_of_tensors.py:12]: Please define PDDs -> PDDs (Principal Diffusion Directions)

File: scripts/scil_mrds_select_number_of_tensors.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Fit the info from a NuFO map to the plausible MRDS solutions.
+
+Each MRDS input is a list of 5 files:
+    - Component size
```
Comment (mdesco) [inline scripts/scil_mrds_select_number_of_tensors.py:8]: Not a fan of component size. You mean signal fraction of the component, is that it?

File: scripts/scil_mrds_select_number_of_tensors.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    iso_out = np.zeros((X, Y, Z, 2))
+    num_tensors_out = np.zeros((X, Y, Z), dtype=np.uint8)
+    evecs_out = np.zeros((X, Y, Z, 9))
+
+    # select data using mosemap
+    for (X, Y, Z) in filtered_voxels:
+        N = mosemap[X, Y, Z]-1
+
+        if N > 2:
+            N = 2
```
Comment (EmmaRenauld) [inline scripts/scil_mrds_select_number_of_tensors.py:139]: Should this be explained somewhere? Nufo values will be clipped? Or is it part of the MRDS theory?

File: scripts/scil_mrds_select_number_of_tensors.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    - Signal fraction of each tensor ([in_prefix]_V[1,2,3]_signal_fraction.nii.gz)
+    - Eigenvalues ($in_prefix]_V[1,2,3]_eigenvalues.nii.gz)
+    - Isotropic ([in_prefix]_V[1,2,3]_isotropic.nii.gz)
+    - Number of tensors ([in_prefix]_V[1,2,3]_num_tensors.nii.gz)
+    - Eigenvectors ([in_prefix]_V[1,2,3]_evecs.nii.gz)
+
+
+    Example:
+        scil_mrds_select_number_of_tensors.py nufo.nii.gz
+            --in_prefix sub-01
```
Comment (EmmaRenauld) [inline scripts/scil_mrds_select_number_of_tensors.py:29]: Actually, in_prefix is a required arg (no `--`)

**Review by EmmaRenauld — APPROVED** [review]

**Pull request merged**